fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexAggregateFunctionCall.java

```diff
@@ -0,0 +1,272 @@
+        this.groupedExpressionOptional = groupedExpressionOptional;
+    }
+
+    @Nonnull
+    public String getName() {
+        return name;
+    }
+
+    @Nonnull
+    public GroupingKeyExpression getGroupingKeyExpression() {
+        return groupingKeyExpression;
+    }
+
+    @Nonnull
+    public Set<KeyExpression> getGroupingExpressions() {
+        return groupingExpressions;
+    }
+
+    @Nonnull
+    public Optional<KeyExpression> getGroupedExpressionOptional() {
+        return groupedExpressionOptional;
+    }
+
+    @Nonnull
+    public KeyExpression getGroupedExpressionOrEmpty() {
+        return groupedExpressionOptional.orElse(EmptyKeyExpression.EMPTY);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof IndexAggregateFunctionCall)) {
+            return false;
+        }
+        final IndexAggregateFunctionCall that = (IndexAggregateFunctionCall)o;
+        return Objects.equal(getName(),
+                that.getName()) &&
+               Objects.equal(getGroupingKeyExpression(), that.getGroupingKeyExpression()) &&
+               Objects.equal(getGroupingExpressions(), that.getGroupingExpressions()) &&
+               Objects.equal(getGroupedExpressionOptional(), that.getGroupedExpressionOptional());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(getName(), getGroupingKeyExpression(), getGroupingExpressions(), getGroupedExpressionOptional());
+    }
+
+    @Nonnull
+    public IndexAggregateFunctionCall withNewExpressions(@Nonnull Iterable<KeyExpression> groupingExpressions,
+                                                         @Nonnull Optional<KeyExpression> groupedExpressionOptional) {
+        return new IndexAggregateFunctionCall(getName(), groupingExpressions, groupedExpressionOptional);
+    }
+
+    @Nonnull
+    public QueryComponent applyCondition(@Nonnull QueryComponent filterCondition) {
+        return filterCondition;
+    }
+
+    /**
+     * Method to create a stream of enumerated {@link IndexAggregateFunction}s that represent the bound versions of this
+     * function call.
+     * In order to bind the function call, we need to associated the aggregate function call with an index and also
+     * pin the order of grouping key expressions in a way that is compatible with that index. Note that the returned
+     * index aggregate functions are merely candidates for actual index-compatible ones as we don't check here the
+     * conditions that constitute a compatible index. On the contrary, this method is intended to be used by the logic
+     * that builds the actual function call for later consumption.
+     * @param indexName indexName to use
+     * @return a {@link Stream} of {@link IndexAggregateFunction}s
+     */
+    public Stream<IndexAggregateFunction> enumerateIndexAggregateFunctionCandidates(@Nonnull String indexName) {
```

Comment:
Should use `index` or `indexName` consistently.

Reply:
changed every signature in this class to use  to `indexName`

Reply:
✅